catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/changeProjectOrderTest.java

```diff
@@ -0,0 +1,170 @@
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.io.XstreamSerializer;
+import org.catrobat.catroid.ui.ProjectListActivity;
+import org.catrobat.catroid.uiespresso.testsuites.Cat;
+import org.catrobat.catroid.uiespresso.testsuites.Level;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.catrobat.catroid.utils.FileMetaDataExtractor;
+import org.catrobat.catroid.utils.PathBuilder;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.security.spec.ECField;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import static android.support.test.InstrumentationRegistry.getInstrumentation;
+import static android.support.test.espresso.Espresso.closeSoftKeyboard;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.openActionBarOverflowOrOptionsMenu;
+import static android.support.test.espresso.Espresso.pressBack;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.replaceText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.RootMatchers.isDialog;
+import static android.support.test.espresso.matcher.ViewMatchers.hasDescendant;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.isEnabled;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.common.FlavoredConstants.DEFAULT_ROOT_DIRECTORY;
+import static org.catrobat.catroid.uiespresso.ui.fragment.rvutils.RecyclerViewInteractionWrapper.onRecyclerView;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.fail;
+
+@RunWith(AndroidJUnit4.class)
+public class changeProjectOrderTest {
+
+	@Rule
+	public BaseActivityInstrumentationRule<ProjectListActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ProjectListActivity.class, true, false);
+
+	private String ProjectOneName = "ProjectOneName";
+	private String ProjectTwoName = "ProjectTwoName";
+
+	@Before
+	public void setUp() throws Exception {
+		createProject(ProjectOneName);
+		createProject(ProjectTwoName);
+
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void changeProjectOrderTest() {
+		try{
+			Thread.sleep(2000);
+		}
+		catch (Exception e){}
+		onRecyclerView().performOnItemWithText(ProjectTwoName,click());
```

Comment:
please no switching activities in espresso tests if it is not really necessary / testable in another way.
alternatively you could just directly change the "lastModified" timestamp in a project and trigger a refresh of the adapter.

Reply:
Purpose of the test is to check that the transition from list to a project changes the projects last modified .
If I change the lastModified timestamp in the test itself and someone deletes the code that changes lastModified on opening a project, the test would still pass.